### PR TITLE
Add Python 3.10 support

### LINKED ISSUE #1270: Python 3.10 support
### Use Case/Motivation

Python 3.10.0 beta 3 is available for testing, and Fedora Linux has already integrated Python 3.10 in the development version (Rawhide). Fedora 35 will release with Python 3.10 in the fall, around the same time that Python 3.10 final is available.

As the maintainer of the `jrnl` package in Fedora, I’m loosening the [Python version restriction in pyproject.toml](https://github.com/jrnl-org/jrnl/blob/v2.8.1/pyproject.toml#L30) downstream.

With a recent update to [asteval](https://github.com/newville/asteval/releases), jrnl now builds, the tests pass (`pytest` and `behave`), and the executable passes a basic interactive “smoke test” at the command line.

Since it seems there are likely no changes required other than `pyproject.toml`, please consider officially supporting Python 3.10 at your earliest convenience.

### Example Usage

(not applicable)

### Other information

`pytest` results:
```
=============================== warnings summary ===============================
../../../../usr/lib/python3.10/site-packages/ansiwrap/core.py:6
  /usr/lib/python3.10/site-packages/ansiwrap/core.py:6: DeprecationWarning: the imp module is deprecated in favour of importlib and slated for removal in Python 3.12; see the module's documentation for alternative uses
    import imp

-- Docs: https://docs.pytest.org/en/stable/warnings.html
======================== 96 passed, 1 warning in 0.87s =========================
```

`behave` results:
```
16 features passed, 0 failed, 0 skipped
305 scenarios passed, 0 failed, 5 skipped
1683 steps passed, 0 failed, 27 skipped, 0 undefined
Took 0m27.708s
```

### PR DESCRIPTION
**Pull request by micahellison**
Fixes #1270, if all the tests pass on all platforms.

### Checklist

- [x] I have read the [contributing doc](https://github.com/jrnl-org/jrnl/blob/develop/CONTRIBUTING.md).
- [x] I have included a link to the relevant issue number.
- [ ] I have tested this code locally.
- [x] I have checked to ensure there aren't other open [pull requests](../pulls)
  for the same issue.
- [ ] I have written new tests for these changes, as needed.
- [x] All tests pass.

### CODE REVIEW

File: .github/workflows/testing.yaml [hunk 1]
```diff
@@ -25,11 +25,8 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
```
Comment (wren) [inline .github/workflows/testing.yaml:27]:
```suggestion
    strategy:
      fail-fast: false
```

File: .github/workflows/testing.yaml [hunk 2]
```diff
@@ -25,11 +25,8 @@ jobs:
     runs-on: ${{ matrix.os }}
```
Comment (wren) [inline .github/workflows/testing.yaml:25]:
```suggestion
    runs-on: ${{ matrix.os }}
    continue-on-error: ${{ matrix.python-version == '3.10-dev' }}
```

**Comment by musicinmybrain** [comment]
I managed to reproduce the CI failure in an Ubuntu VM. I also found a report at https://bugzilla.redhat.com/show_bug.cgi?id=1970361, the corresponding upstream bug (https://github.com/python-poetry/poetry/issues/3897), and the PR that fixed the bug: https://github.com/python-poetry/poetry/pull/3895. The PR was merged into master, but it seems it has not been backported to the `1.1` branch, so the bug is still present in the latest release of `poetry`.

**Comment by wren** [comment]
In that case, we could probably install poetry 1.2.0a1 to get through this. I tested locally, and that seems to fix the error. 

I'm hesitant to use an alpha build for our entire pipeline, though, but I'd be okay with doing it just for the python 3.10-dev tests.

**Comment by wren** [comment]
Well, this isn't great. I ran into python-poetry/poetry#3153 while updating our dependencies. I think we might have to install poetry v1.2.0a1 for all of our tests, or I'm afraid they might not run anymore. I did it and now all of the tests are passing.

Thoughts on using an alpha build for our entire pipeline?

**Comment by micahellison** [comment]
Turns out these GitHub Actions failures were a fluke - we reverted and ran again and everything passes.

**Review by wren — APPROVED** [review]
lgtm 👩‍⚕️

**Pull request merged**